src/fsipy/automatedPostprocessing/postprocessing_h5py/postprocessing_common_h5py.py

```diff
@@ -0,0 +1,470 @@
+    """
+    Get coordinates from a mesh file.
+
+    Args:
+        mesh_path (Union[str, Path]): Path to the mesh file.
+
+    Returns:
+        np.ndarray: Array containing the coordinates.
+    """
+    mesh = h5py.File(mesh_path, "r")
+    coords = mesh['mesh/coordinates'][:, :]
+    return coords
+
+
+def get_surface_topology_coords(out_file: Union[str, Path]) -> tuple:
+    """
+    Get surface topology and coordinates from an output file.
+
+    Args:
+        out_file (Union[str, Path]): Path to the output file.
+
+    Returns:
+        tuple: Tuple containing the surface topology and coordinates.
+    """
+    mesh = h5py.File(out_file, "r")
+    topology = mesh["Mesh/0/mesh/topology"][:, :]
+    coords = mesh["Mesh/0/mesh/geometry"][:, :]
+    return topology, coords
+
+
+def get_domain_topology(mesh_path: Union[str, Path]) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Obtain the topology for the fluid, solid, and all elements of the input mesh.
+
+    Args:
+        mesh_path (Union[str, Path]): Path to the HDF5 mesh file.
+
+    Returns:
+        Tuple[np.ndarray, np.ndarray, np.ndarray]: Tuple containing three NumPy arrays:
+            fluid_topology: Topology array for fluid elements.
+            wall_topology: Topology array for solid (wall) elements.
+            all_topology: Topology array for all elements.
+    """
+    vector_data = h5py.File(mesh_path, "r")
+
+    # Open domain array
+    domains_loc = 'domains/values'
+    domains = vector_data[domains_loc][:]
+
+    # Find indices for solid and fluid domains
+    id_wall = np.where(domains > 1)[0]  # domain = 2 is the solid
+    id_fluid = np.where(domains == 1)[0]  # domain = 1 is the fluid
+
+    # Get topology arrays
+    topology_loc = 'domains/topology'
+    all_topology = vector_data[topology_loc][:, :]
+    wall_topology = all_topology[id_wall, :]
+    fluid_topology = all_topology[id_fluid, :]
+
+    return fluid_topology, wall_topology, all_topology
+
+
+def get_domain_ids(mesh_path: Union[str, Path]) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Obtain a list of the node IDs for the fluid, solid, and all elements of the input mesh.
+
+    Args:
+        mesh_path (Union[str, Path]): Path to the HDF5 mesh file.
+
+    Returns:
+        Tuple[np.ndarray, np.ndarray, np.ndarray]: Tuple containing three NumPy arrays:
+            fluid_ids: Node IDs for fluid elements, sorted in ascending order.
+            wall_ids: Node IDs for solid (wall) elements, sorted in ascending order.
+            all_ids: Node IDs for all elements, sorted in ascending order.
+    """
+    fluid_topology, wall_topology, all_topology = get_domain_topology(mesh_path)
+
+    # Find the unique node ids in the wall, fluid, and all topologies, sorted in ascending order
+    wall_ids = np.unique(wall_topology)
+    fluid_ids = np.unique(fluid_topology)
+    all_ids = np.unique(all_topology)
+
+    return fluid_ids, wall_ids, all_ids
```

Comment:
I implemented a function for getting IDs. This new version is basically combined version of old `get_domain_topology` and `get_domain_ids`. Do you think it is worth moving new function so that it can be used in `postprocessing_h5py` as well?

Reply:
Here is the link to the function https://github.com/KVSlab/Aneurysm_Workflow_FSI/blob/d98e4c73b9136592df3b06214a1b31ee2ee31b58/src/fsipy/automatedPostprocessing/postprocessing_fenics/create_hdf5.py#L174-L210